selfdrive/locationd/paramsd.py

```diff
@@ -161,7 +161,7 @@ def main(sm=None, pm=None):
   while True:
     sm.update()
     for which in sorted(sm.updated.keys(), key=lambda x: sm.logMonoTime[x]):
-      if sm.updated[which]:
+      if sm.updated[which] and sm.all_alive_and_valid():
```

Comment:
do you want to throw away the valid stuff? you can pass `all_alive_and_valid` a list with a subset of messages if you want like this
```suggestion
      if sm.updated[which] and sm.all_alive_and_valid([which]):
```

Reply:
I want to update the filter when all the required messages are alive and valid. The current issue is when the sensor messages are valid and alive, but the camera odometry messages take a while (~2 secs) during init. The filter then enters a bad state and recovers slowly.

Reply:
I would check this outside the loop in that case